onebusaway-android/src/main/java/org/onebusaway/android/ui/ArrivalsListLoader.java

```diff
@@ -51,11 +51,23 @@ public ArrivalsListLoader(Context context, String stopId) {
 
     @Override
     public ObaArrivalInfoResponse loadInBackground() {
-        ObaArrivalInfoRequest obaArrivalInfoRequest = ObaArrivalInfoRequest.newRequest(getContext(),
-                mStopId, mMinutesAfter);
-        // Cache the URL so we have a record of the request w/ params made to the server
-        mUrl = obaArrivalInfoRequest.getUri().toString();
-        return obaArrivalInfoRequest.call();
+        ObaArrivalInfoResponse response;
+
+        do {
```

Comment:
We can do something better instead of hitting the server multiple times @aaronbrethorst :thinking:

Reply:
This is actually the same solution I came up with for iOS :)

Reply:
I was thinking that we could directly load the arrival buses instead of having to click "Load More" repeatedly.

Reply:
I will test that and get back to you